Extension/src/Utility/System/guards.ts

```diff
@@ -0,0 +1,67 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
```

Comment:
Why is this file called "guard" -- is there a better name that could be used? Should it be called "is.ts"?

Reply:
These kinds of functions are referred to as type guards, so the name actually works.

https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
But is "guard" too general? Could the file be typeGuard.ts so that's more descriptive?

Reply:
I'm happy with guard.